packages/frontend-api/src/Resources/config/graphql-types/AdvertImageDecorator.types.yaml

```diff
@@ -0,0 +1,23 @@
+AdvertImageDecorator:
+    type: object
+    decorator: true
+    inherits:
+        - 'AdvertDecorator'
+    config:
+        fields:
+            link:
+                type: "String"
+                description: "Advert link"
+            image:
+                type: "Image"
+                description: "Advert image"
+                resolve: '@=service("Shopsys\\FrontendApiBundle\\Model\\Resolver\\Image\\ImagesResolver").resolveByAdvert(value, args["type"], args["size"])'
+                args:
+                    type:
+                        type: "String"
+                        defaultValue: null
+                    size:
+                        type: "String"
+                        defaultValue: null
```

Comment:
what's the difference between image in adverts and for example brand?
brand has the image field defined as an array, advert do not. 

I cannot query original size of the image the same way I can with brand.

_I realize positions placed as sizes in advert image config are extremely weird, but I would rather see consistent API_

Reply:
solved in 5b26911377eaae0cb10533f3b7c57aeb84856904